x-pack/beatless/provider/aws/cloudwatch_logs.go

```diff
@@ -0,0 +1,183 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package aws
+
+import (
+	"context"
+	"errors"
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatchlogs"
+	lambdaApi "github.com/aws/aws-sdk-go-v2/service/lambda"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/x-pack/beatless/core"
+	"github.com/elastic/beats/x-pack/beatless/provider"
+	"github.com/elastic/beats/x-pack/beatless/provider/aws/transformer"
+)
+
+// CloudwatchLogsConfig is the configuration for the cloudwatchlogs event type.
+type CloudwatchLogsConfig struct {
+	Triggers    []*CloudwatchLogsTriggerConfig
+	Description string `config:"description"`
+	Name        string `config:"name" validate:"nonzero,required"`
+	Role        string `config:"role" validate:"nonzero,required"`
+}
+
+// CloudwatchLogsTriggerConfig is the configuration for the specific triggers for cloudwatch.
+type CloudwatchLogsTriggerConfig struct {
+	LogGroupName  string `config:"log_group_name" validate:"nonzero,required"`
+	FilterName    string `config:"filter_name" validate:"nonzero,required"`
+	FilterPattern string `config:"filter_pattern"`
+}
+
+// Validate validates the configuration.
+func (cfg *CloudwatchLogsConfig) Validate() error {
+	if len(cfg.Triggers) == 0 {
+		return errors.New("you need to specify at least one trigger")
+	}
+	return nil
+}
+
+// CloudwatchLogs receives CloudwatchLogs events from a lambda function and forward the logs to
+// an Elasticsearch cluster.
+type CloudwatchLogs struct {
+	log    *logp.Logger
+	config *CloudwatchLogsConfig
+}
+
+// NewCloudwatchLogs create a new function to listen to cloudwatch logs events.
+func NewCloudwatchLogs(provider provider.Provider, cfg *common.Config) (provider.Function, error) {
+	config := &CloudwatchLogsConfig{}
+	if err := cfg.Unpack(config); err != nil {
+		return nil, err
+	}
+	return &CloudwatchLogs{log: logp.NewLogger("cloudwatch_logs"), config: config}, nil
+}
+
+// Run start the AWS lambda handles and will transform any events received to the pipeline.
+func (c *CloudwatchLogs) Run(_ context.Context, client core.Client) error {
+	lambda.Start(func(request events.CloudwatchLogsEvent) error {
+		parsedEvent, err := request.AWSLogs.Parse()
+		if err != nil {
+			c.log.Errorf("could not parse events from cloudwatch logs, error: %s", err)
+			return err
+		}
+
+		// defensive checks
+		if len(parsedEvent.LogEvents) == 0 {
```

Comment:
Why is it invalid to get no events?

Reply:
I've looked at the SDK; this cannot happen the function should not be invoked in that case. I've removed the defensive check for all the function types.